### PR TITLE
Migrate pythonnet to .NET Core 3.0

### PR DESCRIPTION
**Pull request by gsalaz98**
### What does this implement/fix? Explain your changes.

Migrates and adds support for .NET Core 3.0 for pythonnet. This is required for python algorithms to run in LEAN .NET Core

### Does this close any currently open issues?

No

### Any other comments?

I believe this might break backwards compatibility with .NET Framework 4.5.2 -- which is what LEAN currently runs on. Will have to do some testing for that.

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [ ] If an enhancement PR, please create docs and at best an example
-   [ ] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

File: src/coreclr/pynetinit.c [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+    }
+#endif
+
+    if (!GetEntrypointExecutableAbsolutePath(&pn_args->entry_path))
+    {
+        pn_args->error = "Unable to find entry point";
+        return;
+    }
+
+    if (!GetClrFilesAbsolutePath(pn_args->entry_path, "/home/gerry/dotnet/shared/Microsoft.NETCore.App/2.0.0", &pn_args->clr_path))
```
Comment (gsalaz98) [inline src/coreclr/pynetinit.c]: We need to figure out a way to remove the hard coded path from here. I believe that we can pass NULL as the parameter value and it'll try to determine the path automatically.

File: setup.py [hunk 2]
```diff
@@ -22,10 +22,10 @@
 # Allow config/verbosity to be set from cli
 # http://stackoverflow.com/a/4792601/5208670
 CONFIG = "Release"  # Release or Debug
-VERBOSITY = "minimal"  # quiet, minimal, normal, detailed, diagnostic
+VERBOSITY = "normal"  # quiet, minimal, normal, detailed, diagnostic
```
Comment (gsalaz98) [inline setup.py]: Reset verbosity back to "minimal"?

File: src/clrmodule.csproj [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (gsalaz98) [inline src/clrmodule.csproj]: This file might not be needed

File: setup.py (hunk 2 above)
Comment (AlexCatarino) [reply setup.py]: I don't see a good reason to change it.

**Comment by Martin-Molinero** [comment]
Closed in favor of https://github.com/QuantConnect/pythonnet/pull/47

**Pull request closed without merging**